src/transactions/SponsorshipUtils.cpp

```
@@ -131,16 +131,34 @@ isSignerSponsored(std::vector<Signer>::const_iterator const& signerIt,
 //
 ////////////////////////////////////////////////////////////////////////////////
 static uint32_t
-computeMultiplier(LedgerEntryType let)
+computeMultiplier(LedgerEntry const& le)
 {
-    // TODO(jonjove): Handle ClaimableBalance
-    if (let == ACCOUNT)
+    auto type = le.data.type();
+    if (type == ACCOUNT)
     {
         return 2;
     }
+    else if (type == CLAIMABLE_BALANCE)
+    {
+        return le.data.claimableBalance().claimants.size();
+    }
+
     return 1;
 }
 
+static bool
+isSubentry(LedgerEntry const& le)
+{
+    switch (le.data.type())
+    {
+    case ACCOUNT:
+    case CLAIMABLE_BALANCE:
+        return false;
+    default:
```

Nitpick: This presents a minor danger for future maintainers. Might be better to be super explicit here (eg. add cases for the other ledger entry types) and then let the default throw.

---

This is a good point. Updated.